packages/api/lambdas/process-s3-dead-letter-archive.js

```diff
@@ -3,12 +3,69 @@
 const log = require('@cumulus/common/log');
 
 const { s3 } = require('@cumulus/aws-client/services');
-const { getJsonS3Object } = require('@cumulus/aws-client/S3');
+const { getJsonS3Object, deleteS3Object } = require('@cumulus/aws-client/S3');
+const S3 = require('@cumulus/aws-client/S3');
 const { getKnexClient } = require('@cumulus/db');
 const { unwrapDeadLetterCumulusMessage } = require('@cumulus/message/DeadLetterMessage');
 
 const { writeRecords } = require('./sf-event-sqs-to-db-records');
 
+/**
+ * Returns string of ISO Date Year/Month/Day
+ *
+ * @param {Date}   [date] - date
+ * @returns {string}
+ */
+const getISODate = (date = new Date()) => `${date.getFullYear()}-${date.getMonth()}-${date.getDate()}`;
+
+/**
+ * Generates new archive key for unprocessed dead letter message
+ *
+ * @param {string}   [failedKey] - key of message that failed to process
+ * @returns {string}
+ */
+const generateNewArchiveKeyForFailedMessage = (failedKey) => {
+  // Split key with format `${process.env.stackName}/dead-letter-archive/sqs/${messageId}`,
```

Comment:
1. `<stackName>/dead-letter-archive/sqs/` is the default location, but not necessarily the path from the request. https://nasa.github.io/cumulus-api/#recover-cumulus-messages
2. If you put the failed message to `${pathPrefix}/sqs/${getISODate()}/${messageId}`, it will still be processed again, since it's under the same prefix as `<stackName>/dead-letter-archive/sqs/`

Reply:
https://github.com/nasa/cumulus/pull/3099/commits/5119f6385dc9b7ce474429112fe808ba6f969d7e

Reply:
If the `message` is not in the default location 
https://github.com/nasa/cumulus/blob/d0da1d9358e9ec5b894ceb75fc37f21c86acbe02/packages/api/lambdas/process-s3-dead-letter-archive.js#L84
You may not able to split path with `/sqs/`, I suggest to put failed message to default location:
`${process.env.stackName}/dead-letter-archive/failed-sqs/$date`
Or we can take  something like `failedpath` from user request, but that requires more work